core/src/main/kotlin/live/LiveGuild.kt

```diff
@@ -19,6 +20,124 @@ import dev.kord.core.event.user.VoiceStateUpdateEvent
+
+@KordPreview
+fun LiveGuild.onRoleUpdate(block: suspend (RoleUpdateEvent) -> Unit) = on(consumer = block)
+
+@KordPreview
+fun LiveGuild.onRoleDelete(block: suspend (RoleDeleteEvent) -> Unit) = on(consumer = block)
+
+@KordPreview
+fun LiveGuild.onMemberJoin(block: suspend (MemberJoinEvent) -> Unit) = on(consumer = block)
+
+@KordPreview
+fun LiveGuild.onMemberUpdate(block: suspend (MemberUpdateEvent) -> Unit) = on(consumer = block)
+
+@KordPreview
+fun LiveGuild.onMemberLeave(block: suspend (MemberLeaveEvent) -> Unit) = on(consumer = block)
+
+@KordPreview
+inline fun LiveGuild.onReaction(crossinline block: suspend (Event) -> Unit) = on<Event> {
+    if (it is ReactionAddEvent || it is ReactionRemoveEvent) {
+        block(it)
+    }
+}
+
+@KordPreview
+inline fun LiveGuild.onReaction(
+    emoji: ReactionEmoji,
+    crossinline block: suspend (Event) -> Unit
+) = on<Event> {
+    if (it is ReactionAddEvent && (emoji == it.emoji) || it is ReactionRemoveEvent && (emoji == it.emoji)) {
+        block(it)
+    }
+}
+
+@KordPreview
+fun LiveGuild.onReactionAdd(block: suspend (ReactionAddEvent) -> Unit) = on(consumer = block)
+
+@KordPreview
+inline fun LiveGuild.onReactionAdd(
+    reaction: ReactionEmoji,
+    crossinline block: suspend (ReactionAddEvent) -> Unit
+) = on<ReactionAddEvent> {
+    if (it.emoji == reaction) {
+        block(it)
+    }
+}
+
+@KordPreview
+fun LiveGuild.onReactionRemove(block: suspend (ReactionRemoveEvent) -> Unit) = on(consumer = block)
+
+@KordPreview
+inline fun LiveGuild.onReactionRemove(
+    reaction: ReactionEmoji,
+    crossinline block: suspend (ReactionRemoveEvent) -> Unit
+) = on<ReactionRemoveEvent> {
+    if (it.emoji == reaction) {
+        block(it)
+    }
+}
+
+@KordPreview
+fun LiveGuild.onReactionRemoveAll(block: suspend (ReactionRemoveAllEvent) -> Unit) = on(consumer = block)
+
+@KordPreview
+fun LiveGuild.onMessageCreate(block: suspend (MessageCreateEvent) -> Unit) = on(consumer = block)
+
+@KordPreview
+fun LiveGuild.onMessageUpdate(block: suspend (MessageUpdateEvent) -> Unit) = on(consumer = block)
+
+@KordPreview
+fun LiveGuild.onMessageDelete(block: suspend (MessageDeleteEvent) -> Unit) = on(consumer = block)
+
+@KordPreview
+fun LiveGuild.onChannelCreate(block: suspend (ChannelCreateEvent) -> Unit) = on(consumer = block)
+
+@KordPreview
+fun LiveGuild.onChannelUpdate(block: suspend (ChannelUpdateEvent) -> Unit) = on(consumer = block)
+
+@KordPreview
+fun LiveGuild.onChannelDelete(block: suspend (ChannelDeleteEvent) -> Unit) = on(consumer = block)
+
+@KordPreview
+fun LiveGuild.onGuildCreate(block: suspend (GuildCreateEvent) -> Unit) = on(consumer = block)
+
+@KordPreview
+fun LiveGuild.onGuildUpdate(block: suspend (GuildUpdateEvent) -> Unit) = on(consumer = block)
+
+@KordPreview
+fun LiveGuild.onGuildDelete(block: suspend (GuildDeleteEvent) -> Unit) = on(consumer = block)
+
+@KordPreview
+fun LiveGuild.onShutDown(block: suspend (GuildDeleteEvent) -> Unit) = onGuildDelete(block)
```

Comment:
Yes i understand, in this case for the LiveEntity, maybe override ShutDown do call a similar function :thinking:  ?

Reply:
This sort of moves the live entity from being a glorified event filter with a variable to a real object with live states and separate listeners. Which I guess is fine since we'll have to make live entities something more grand to validate their existence in the first place. Sure, this seems like a good idea.

Reply:
So far I have just deleted this method. To see later for a more concrete implementation maybe